tests/Oryx.Integration.Tests/Node/NodeWebFrontEndAppTest.cs

```diff
@@ -62,5 +63,93 @@ public async Task CanBuildAndRun_NodeWebFrontEndApp(string nodeVersion)
                     Assert.Contains("Say It Again", data);
                 });
         }
+
+        [Theory]
+        [InlineData("webfrontend")]
+        [InlineData("webfrontend-yarnlock")]
+        public async Task CanBuildAndRun_NodeWebFrontEndApp_WhenPruneDevDependenciesIsTrue(string appName)
+        {
+            // Arrange
+            var nodeVersion = "10";
+            var volume = CreateAppVolume(appName);
+            var appDir = volume.ContainerDir;
+            var buildScript = new ShellScriptBuilder()
+               .AddCommand(
+                $"oryx build {appDir} --platform nodejs --platform-version {nodeVersion} " +
+                $"-p {NodePlatform.PruneDevDependenciesPropertyKey}=true")
+               .ToString();
+            var runScript = new ShellScriptBuilder()
+                .AddCommand($"oryx -appPath {appDir} -bindPort {ContainerPort}")
+                .AddCommand(DefaultStartupFilePath)
+                .ToString();
+
+            await EndToEndTestHelper.BuildRunAndAssertAppAsync(
+                appName,
+                _output,
+                volume,
+                "/bin/sh",
+                new[]
+                {
+                    "-c",
+                    buildScript
+                },
+                $"oryxdevmcr.azurecr.io/public/oryx/node-{nodeVersion}",
+                ContainerPort,
+                "/bin/sh",
+                new[]
+                {
+                    "-c",
+                    runScript
+                },
+                async (hostPort) =>
+                {
+                    var data = await _httpClient.GetStringAsync($"http://localhost:{hostPort}/");
+                    Assert.Contains("Say It Again", data);
+                });
+        }
+
+        [Fact]
+        public async Task CanBuildAndRun_NodeWebFrontEndApp_AfterRebuild_WhenPruneDevDependenciesIsTrue()
```

Comment:
I would also add a test where NodePlatform.PruneDevDependenciesPropertyKey=false/blank just to ensure we are not break anything currently working

Reply:
Sure, sounds good. added one now.